noloco/utils.py

```diff
@@ -247,10 +246,23 @@ def options_without_data(options):
 
     return new_options
 
+
 def result_name_suffix(query):
     if query == 'findMany':
         return 'Collection'
     elif query == 'findUnique':
         return ''
     else:
         raise NolocoQueryNotSupportedError(query)
+
+
+def reverse_name_matches_relationship_name(field, relationship_name):
+    reverse_name = field['reverseName']
+
+    if reverse_name is None or reverse_name == '':
+        return False
+
+    reverse_collection_name = field['reverseName'] + COLLECTION
```

Comment:
If it's 1:1 we wouldn't acctually append `COLLECTION` so you might need to adjust this logic slightly

Reply:
That's fair right now I was trying to just catch both cases in one go but I can pivot on whether it is a collection or not.